### DISCUSSION TITLE
Some questions about test running in nvim 0.10 and the plugin configuration in general

### DISCUSSION THREAD

**Discussion by x1a0** [mrcjkb/rustaceanvim#233] [Q&A] [answered]
Thanks for this great plugin!

Since upgraded NeoVim to 0.10, I noticed that when running all tests (via `runnables`), it does not show the split window any more. This seems due to in 0.10 the test executor is default to `background`.
Hence I have 3 questions:
- To bring back the "old" way, should I set `test_executor = executors.termopen`?
- If above is correct, where should I set it? Based on README I think I need to have this somewhere:
  ```
  vim.g.rustaceanvim = {
    tools = {
      test_executor = executors.termopen
    }
  }
  ```
  [In the README it says](https://github.com/mrcjkb/rustaceanvim/blob/master/README.md?plain=1#L652-L655):
  > The options shown below are the defaults. You only need to pass the keys to the setup function that you want to be changed, because the defaults are applied for keys that are not provided.
  
  But in the README it also emphasizes that `setup` should not be called. So I am a bit confused that where should I put this table so it'd be correctly loaded. (I use Lazy btw).
- And another question is, how can I configure it so the testables/runnables executions run in a split window to the right (so a vertical split)? They are shown in the top window in a horizontal split by default it seems.

Thanks again!

**Comment by mrcjkb** [comment]
Hey 👋 

That quote from the readme was outdated rust-tools documentation that I forgot to remove. This plugin doesn't have a `setup` function.

You can set the `test_executor` using a string alias:

```lua
vim.g.rustaceanvim = {
  tools = {
    test_executor = 'termopen',
  },
}
```

It doesn't really matter where you put the table. This plugin will read from it when you open a rust file.

lazy.nvim users typically define it in the `init` or `config` function of the plugin spec.

Currently, horizontal splitting for `termopen` is hard coded, I'm afraid. But PRs are welcome 😀
You can also open a feature request, but I might need some time to implement it, as I don't use the `termopen` executor myself.

**Reply by x1a0** [reply]
Thanks! I put it in `config` now and everything seems good 👍 

Regarding the `background` mode (or maybe about "executions" in general), if say something is running in background, is it possible to "attach to it by split a window"? For example if I invoke a `runnable`, and press `ECS` while the cursor is in the runnable window, it'd close the window, but the process is still running in background. In my case it's a http server so it'd still be occupying the port. I realized that I could re-run it so it'd kill the prev process and start a new one. However I wonder is it possible to attach to the old process again somehow.

**Reply by mrcjkb** [reply, marked as answer]
The `background` `test_executor` is only intended for tests, not `runnables` in general.
With the `termopen` executor, I don't think you can reattach (unless the buffer is still open, but hidden, in which case it would be a question of finding and switching back to the buffer).
The `toggleterm` executor  (requires the [toggleterm plugin](https://github.com/akinsho/toggleterm.nvim)) should be able to reattach, using the `:Toggleterm <id>` command, where `<id>` is the ID of the terminal (numbered starting from 1).

**Reply by mrcjkb** [reply]
Come to think of it (I'm not 100 % sure), it may be able to configure toggleterm in such a way that this plugin's toggleterm executor opens in a vertical split.
